packages/datadog-plugin-child_process/src/scrub-cmd-params.js

```diff
@@ -0,0 +1,125 @@
+'use strict'
+
+const shellParser = require('shell-quote/parse')
+
+const ALLOWED_ENV_VARIABLES = ['LD_PRELOAD', 'LD_LIBRARY_PATH', 'PATH']
+const PROCESS_DENYLIST = ['md5']
+
+const VARNAMES_REGEX = /\$([\w\d_]*)(?:[^\w\d_]|$)/gmi
+// eslint-disable-next-line max-len
+const PARAM_PATTERN = '^-{0,2}(?:p(?:ass(?:w(?:or)?d)?)?|api_?key|secret|a(?:ccess|uth)_token|mysql_pwd|credentials|(?:stripe)?token)$'
+const regexParam = new RegExp(PARAM_PATTERN, 'i')
+const ENV_PATTERN = '^(\\w+=\\w+;)*\\w+=\\w+;?$'
+const envvarRegex = new RegExp(ENV_PATTERN)
+const REDACTED = '?'
+
+function extractVarNames (expression) {
+  const varNames = new Set()
+  let match
+
+  while ((match = VARNAMES_REGEX.exec(expression))) {
+    varNames.add(match[1])
+  }
+
+  const varNamesObject = {}
+  for (const varName of varNames.keys()) {
+    varNamesObject[varName] = `$${varName}`
+  }
+  return varNamesObject
+}
+
+function getTokensByExpression (expressionTokens) {
+  const expressionListTokens = []
+  let wipExpressionTokens = []
+  let isNewExpression = true
+
+  expressionTokens.forEach(token => {
+    if (isNewExpression) {
+      expressionListTokens.push(wipExpressionTokens)
+      isNewExpression = false
+    }
+
+    wipExpressionTokens.push(token)
+
+    if (token.op) {
+      wipExpressionTokens = []
+      isNewExpression = true
+    }
+  })
+  return expressionListTokens
+}
+
+function scrubChildProcessCmd (expression) {
+  const varNames = extractVarNames(expression)
+  const expressionTokens = shellParser(expression, varNames)
+
+  const expressionListTokens = getTokensByExpression(expressionTokens)
+
+  const result = []
+  expressionListTokens.forEach((expressionTokens) => {
+    let foundBinary = false
+    for (let index = 0; index < expressionTokens.length; index++) {
+      const token = expressionTokens[index]
+
+      if (typeof token === 'object') {
+        if (token.pattern) {
+          result.push(token.pattern)
+        } else if (token.op) {
+          result.push(token.op)
+        } else if (token.comment) {
+          result.push(`#${token.comment}`)
+        }
+      } else if (!foundBinary) {
+        if (envvarRegex.test(token)) {
+          const envSplit = token.split('=')
+
+          if (!ALLOWED_ENV_VARIABLES.includes(envSplit[0])) {
+            envSplit[1] = REDACTED
+
+            const newToken = envSplit.join('=')
+            expressionTokens[index] = newToken
+
+            result.push(newToken)
+          } else {
+            result.push(token)
+          }
+        } else {
+          foundBinary = true
+          result.push(token)
+
+          if (PROCESS_DENYLIST.includes(token)) {
+            for (index++; index < expressionTokens.length; index++) {
```

Comment:
is this correct -> `for (index++; index < expressionTokens.length; index++) {`?

Reply:
It looks intentional but this whole function is difficult to read tbh :/

Reply:
it is intentional, because this code want to scrub all tokens after `md5`
Could a comment help here?
```suggestion
            // Remove every parameter after DENIED token 
            for (index++; index < expressionTokens.length; index++) {
```